### ISSUE TITLE
🐛 BUG: Inconsistent searchParams handling

### ISSUE THREAD

**Issue by htunnicliff** [withastro/astro#3057]
### What version of `astro` are you using?

1.0.0-beta.5

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

## Summary

When SSR is enabled, all odd-numbered query parameters are removed from the URL.

For example, visiting this page

```
http://localhost:3000/?a=odd&b=even&c=odd&d=even&e=odd&f=even
```

will produce the following when logging out the Astro URL:

```js
console.log(Astro.request.url);
// Output: http://localhost:3000/?b=even&d=even&f=even
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/astro-search-params-bug?file=src/pages/index.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by FredKSchott** [comment]
You should no longer being using `experiments.ssr` directly, the new API is exposed by adding an adapter of your choice. We should remove this `experiments.ssr` API now that it's been released out of experiments mode in the v1.0 beta.

Here's an updated SSR example that is working: https://stackblitz.com/edit/astro-search-params-bug-thyjew?file=astro.config.mjs

**Comment by FredKSchott** [comment]
Actually, you were hitting a bug as well. Filed and fixed in #3066

**Comment by htunnicliff** [comment]
Thanks @FredKSchott!
